packages/ts/react-crud/src/autogrid.tsx

```diff
@@ -23,16 +23,84 @@ import type PropertyStringFilter from './types/dev/hilla/crud/filter/PropertyStr
 import type Sort from './types/dev/hilla/mappedtypes/Sort';
 import Direction from './types/org/springframework/data/domain/Sort/Direction';
 
+type AllPropertyNames<TItem> = string & keyof TItem;
```

Comment:
The new types slow down type-checking a lot (testing in IntelliJ). It's several times slower than without, regardless what you touch. 

Also affects auto-complete suggestions. The new types allow providing suggestions for which columns you can use, but it takes several seconds to calculate them. If you just force IntelliJ to auto-complete without waiting for several seconds after typing, it just shows some random stuff.


https://github.com/vaadin/hilla/assets/357820/6553d845-e7c9-4d5d-b309-90449e28d082

Reply:
I have limited inheritance to 3 levels deep, did it help?

Reply:
Performance is a lot better now, the auto-complete as well. Now it starts suggesting sensible options right after typing the first letter of a name.